[Chen Shi Yao, Stephanie] iP

stephanie-csy opened the pull request:
# TALKY MCTALKFACE
**Tired of...**
1. forgetting your tasks? 
2. over-complicated to-do apps? 
3. having no one to talk to?

_Talky McTalkface is here to help!_

> Not having anything on your to do list doesn't mean you don't have anything to do!
> ~ Everyone to me

Hence, Talky McTalkface is here to help you
* Keep track of your todos, events and deadlines
* Remind you to do your work 
* GET YOUR LIFE TOGETHER!

## Features of Talky McTalkface
- [x] add tasks (todos, events and deadlines)
- [x] mark tasks as done
- [x] delete tasks
- [x] save your tasks locally to your computer
- [x] list your events and deadlines on a specific day
- [ ] tell you jokes :wink:
- [ ] sort your tasks

#### Here's the link to my [github](https://github.com/stephanie-csy/ip)
#### And here's a sneak peak of my code!

```javascript
public static void main(String[] args) {
    new Duke("dukeTasks.txt").run(); 
}
```

Commit:
Add Gradle support

stephanie-csy committed:
level-1

stephanie-csy committed:
level-2

stephanie-csy committed:
level-3

stephanie-csy committed:
level-4

stephanie-csy committed:
A-TextUiTesting

stephanie-csy committed:
level-5

stephanie-csy committed:
level-6

stephanie-csy committed:
Added save to hard disk function

stephanie-csy committed:
Added LocalDate for task dates

stephanie-csy committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java

stephanie-csy committed:
Added more OOP

stephanie-csy committed:
Organized classes into packages

stephanie-csy committed:
Add JUnit tests to test the behaviour of the code

stephanie-csy committed:
Package the App as a JAR file

stephanie-csy committed:
Add JavaDocs

stephanie-csy committed:
Tweak the code to comply with the coding standard

stephanie-csy committed:
Add find function

stephanie-csy committed:
Merge branch 'branch-A-CodingStandard'

stephanie-csy committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Ui.java

Ruofan1023 reviewed (commented):
Clean code that generally follows the guidelines. Couldn't find any serious violation haha. Good job!

danielsimre commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,56 @@
+ */
+public class Duke {
+    private final Storage storage;
+    private TaskList tasks;
```
Instead of using `tasks` for this variable name, perhaps `taskList` could be used instead? (Since the variable refers to a singular TaskList object rather than a collection of tasks)

stephanie-csy replied:
Actually this entire class was copied exactly from the CS2103T website under the instructions for A-MoreOOP!

danielsimre commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,41 @@
+package duke;
+
+import duke.command.*;
```
Should the command class imports here be explicitly listed instead?

danielsimre commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,141 @@
+     */
+    public void findTasksOnDate(LocalDate date) {
+        List<Task> foundTasks = new ArrayList<>();
+        for (Task task : this.getTasks()) {
```
(Advanced Rule) I noticed that in `TaskList`, `Event`, and `Deadline`, the keyword `this` is used a lot. Perhaps you could only keep usages of `this` when a field is shadowed by a method or constructor parameter and remove any other usage of it?

danielsimre commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,141 @@
+     * Marks a specified task as done.
+     * @param taskNo The number of the task to be marked as done.
+     */
+    public void taskDone(int taskNo) {
```
Maybe you could change the name of this method to `markTask`, since `taskDone` is not in the form of a verb?

danielsimre reviewed (commented):
Good job overall! Other than a few minor errors, your code closely adheres to the coding standard provided.

stephanie-csy committed:
Merge remote-tracking branch 'origin/add-gradle-support' into add-gradle-support

stephanie-csy committed:
Automate project builds using Gradle, fix some coding standard violations

stephanie-csy committed:
Fix coding style violations found using CheckStyle

Ruofan1023 commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,40 @@
+    public boolean isExit() {
+        return false;
+    }
+}
```
Empty line at the end of file

Ruofan1023 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,141 @@
+        List<Task> foundTasks = new ArrayList<>();
+        for (Task task : getTasks()) {
+            if (task instanceof Deadline) {
+                Deadline d = (Deadline) task;
```
consider naming as 'ddl' or 'deadline' instead of 'd' for better clarity.

stephanie-csy committed:
Add GUI

stephanie-csy committed:
Update Gradle file to bundle JavaFX in JAR

stephanie-csy committed:
Add assertions

stephanie-csy committed:
Improve code quality

stephanie-csy committed:
Merge pull request #1 from stephanie-csy/branch-A-Assertions

Add assertions

stephanie-csy committed:
Merge branch 'master' into branch-A-Assertions

stephanie-csy committed:
Merge branch 'master' into branch-A-CodeQuality

stephanie-csy committed:
Merge pull request #2 from stephanie-csy/branch-A-CodeQuality

Improve code quality

stephanie-csy committed:
Merge branch 'master' into branch-A-Assertions

stephanie-csy committed:
Add ability to recognize and deal with duplicate items

Currently, duplicate tasks are allowed.

Tasks are duplicates when they have the same name (for Todos), or
the same name and date (for Deadlines and Events).

Let's
* check if the task is a duplicate when a user is adding a task
* show a message to let the user know that the task already exists,
  and not add the duplicate task to the task list

stephanie-csy committed:
Add support to edit details of items

Details of items could not be edited after added.

Let's add a new command to allow users to edit details of items
easily after adding them to the task list.

stephanie-csy committed:
Improve GUI

stephanie-csy committed:
Fix style errors and complete JavaDocs

stephanie-csy committed:
Add a User Guide and representative screenshot

stephanie-csy committed:
Set theme jekyll-theme-tactile
